core/camel-core-engine/src/main/java/org/apache/camel/impl/DefaultModel.java

```diff
@@ -96,6 +97,7 @@ public class DefaultModel implements Model {
     private final Map<String, Resilience4jConfigurationDefinition> resilience4jConfigurations = new ConcurrentHashMap<>();
     private final Map<String, FaultToleranceConfigurationDefinition> faultToleranceConfigurations = new ConcurrentHashMap<>();
     private Function<RouteDefinition, Boolean> routeFilter;
+    private RoutesDefinition routeCollection = new RoutesDefinition();
```

Comment:
It is not meant to be shared, I rather prefer that you create a new instance at each call instead.